tox.ini

```diff
@@ -115,3 +115,31 @@ env_dir = {work_dir}/pre-commit
 runner = ignore_env_name_mismatch
 description = Run pre-commit on staged files or arbitrary pre-commit commands (tox run -e pre-commit -- [args])
 commands = pre-commit {posargs:run}
+
+[testenv:clean]
+description = Clean up bytecode and build artifacts
+skip_install = true
+deps = pyclean
+commands = pyclean {posargs:. --debris --erase results/* results/ --yes}
+
+[testenv:ensure_version_matches]
+description = Verify package version is same as Git tag
+deps =
+commands = python -c 'import os; from importlib.metadata import version; pkg, tag = os.environ["PKG_NAME"], os.environ["GIT_TAG"]; ver = version(pkg); error = f"`{ver}` != `{tag}`"; abort = f"Package version does not match the Git tag ({error}). ABORTING."; raise SystemExit(0 if ver and tag and ver == tag else abort)'
```

Comment:
I think it would be cleaner to do something like:
```
commands = python -c 'import os;' \
  'from importlib.metadata import version;' \
  'pkg, tag = os.environ["PKG_NAME"], os.environ["GIT_TAG"];' \
  ...
```
But if this doesn't work for some reason, or there's disagreement on this point, then I'm ok with it as-is.

Reply:
This doesn't work in tox for some weird reasons :-)